Orange/widgets/utils/owlearnerwidget.py

```diff
@@ -164,12 +173,15 @@ def set_data(self, data):
                     "Select one with the Select Columns widget.")
             self.data = None
 
-        self.update_model()
+        # invalidate the model so that handleNewSignals will update it
+        self.model = None
 
-    def apply(self):
+    def apply(self, *, hard=True):
```

Comment:
Updating only when `None` (i.e. no hard update) would be better, but it would break compatibility with derived widgets counting on it.